generator/osm_source.cpp

```diff
@@ -336,10 +337,19 @@ class MainFeaturesEmitter : public EmitterBase
           new feature::FeaturesCollector(info.GetTmpFileName(WORLD_COASTS_FILE_NAME)));
 
     if (info.m_splitByPolygons || !info.m_fileName.empty())
-      m_countries.reset(new TCountriesGenerator(info));
+      m_countries.reset(new TCountriesGenerator(info, m_featureIdToOsmIds));
 
     if (info.m_createWorld)
       m_world.reset(new TWorldGenerator(info));
+
+    string const featureId2OsmIdsFile = info.GetIntermediateFileName(info.m_featureIdToOsmIds, "");
+    LOG(LINFO, ("Saving mapping from feature ids to osm ids to", featureId2OsmIdsFile));
+    m_featureIdToOsmIds.Open(featureId2OsmIdsFile);
+    if (!m_featureIdToOsmIds.IsOpened())
```

Comment:
It would be nice to prepare m_featureIdToOsmIds first, and then initialize m_countries.